contrib/pkg/certificate/create.go

```diff
@@ -0,0 +1,174 @@
+		return err
+	}
+	certbotDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		return fmt.Errorf("cannot create temporary directory for certbot")
+	}
+	log.Infof("Executing certbot command in directory %s", certbotDir)
+	certbotCommand := exec.Command(certbotPath,
+		"certonly",
+		"--non-interactive",
+		"--manual",
+		"--manual-auth-hook", fmt.Sprintf("%s certificate auth-hook --wait-time %v %s %s", os.Args[0], o.WaitTime, o.Region, baseDomainID),
+		"--manual-cleanup-hook", fmt.Sprintf("%s certificate cleanup-hook %s %s", os.Args[0], o.Region, baseDomainID),
+		"--preferred-challenge", "dns",
+		"--config-dir", certbotDir,
+		"--work-dir", certbotDir,
+		"--logs-dir", certbotDir,
+		"--agree-tos",
+		"--manual-public-ip-logging-ok",
+		"--domains", fmt.Sprintf("api.%[1]s.%[2]s,*.apps.%[1]s.%[2]s", o.Name, o.BaseDomain),
+		"--email", o.Email,
+	)
+	buf := &bytes.Buffer{}
+	certbotCommand.Stdout = buf
+	certbotCommand.Stderr = buf
+	err = certbotCommand.Run()
+	if err != nil {
+		fmt.Printf("%s\n", buf.String())
+		return fmt.Errorf("an error occurred creating certificate")
+	}
+
+	certFile := filepath.Join(o.OutputDir, fmt.Sprintf("%s.crt", o.Name))
+	certKey := filepath.Join(o.OutputDir, fmt.Sprintf("%s.key", o.Name))
+
+	err = copyFile(filepath.Join(certbotDir, "live", fmt.Sprintf("api.%s.%s", o.Name, o.BaseDomain), "cert.pem"), certFile)
+	if err != nil {
+		return err
+	}
+	err = copyFile(filepath.Join(certbotDir, "live", fmt.Sprintf("api.%s.%s", o.Name, o.BaseDomain), "privkey.pem"), certKey)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Your certificate has been created successfully.")
+	log.Infof("Letsencrypt directory: %s", certbotDir)
+	log.Infof("Certificate and key are at: %s  %s", certFile, certKey)
+	log.Infof("To create a cluster with these certs:\n    %s create-cluster %s --serving-cert %s --serving-cert-key %s", os.Args[0], o.Name, certFile, certKey)
+	return nil
+}
+
+func (o *Options) getBaseDomainID() (string, error) {
+	client, err := awsclient.NewClient(nil, "", "", o.Region)
+	if err != nil {
+		return "", fmt.Errorf("cannot create AWS client; make sure your environment is setup to communicate with AWS")
+	}
+	result, err := client.ListHostedZonesByName(&route53.ListHostedZonesByNameInput{
+		DNSName: aws.String(o.BaseDomain),
+	})
+	if err != nil {
+		return "", fmt.Errorf("cannot list hosted zones on AWS: %v", err)
+	}
+	hostedZoneID := ""
+	for _, zone := range result.HostedZones {
+		if strings.TrimSuffix(aws.StringValue(zone.Name), ".") == o.BaseDomain {
+			if zone.Config == nil || !aws.BoolValue(zone.Config.PrivateZone) {
+				hostedZoneID = aws.StringValue(zone.Id)
+				break
+			}
+			continue
+		}
+		break
+	}
+	if hostedZoneID == "" {
+		return "", fmt.Errorf("could not find public hosted zone with domain %s", o.BaseDomain)
+	}
+	return hostedZoneID, nil
+}
+
+func copyFile(src, dst string) error {
+	b, err := ioutil.ReadFile(src)
+	if err != nil {
+		return fmt.Errorf("cannot read %s: %v", src, err)
```

Comment:
Since we are already vendoring `github.com/pkg/errors` courtesy of the installer, I would prefer to see us use that package here as follows.
```
return errors.Wrapf(err, "cannot read %s", src)
```